virtualizarr/backend.py

```diff
@@ -1,13 +1,23 @@
+import os
 import warnings
 from collections.abc import Iterable, Mapping
 from enum import Enum, auto
+from functools import partial
 from pathlib import Path
 from typing import (
+    TYPE_CHECKING,
     Any,
+    Callable,
+    Literal,
     Optional,
+    Sequence,
+    cast,
 )
 
-from xarray import Dataset, Index
+from xarray import DataArray, Dataset, Index, combine_by_coords
+from xarray.backends.api import _multi_file_closer
+from xarray.backends.common import _find_absolute_paths
+from xarray.core.combine import _infer_concat_order_from_positions, _nested_combine
```

Comment:
I don't like importing these deep xarray internals like this (though `_infer_concat_order_from_positions` and `_nested_combine` haven't changed since [I wrote them 6 years ago](https://github.com/pydata/xarray/pull/2553)), but the only alternative would be to make a general virtualizarr backend engine for xarray (see https://github.com/zarr-developers/VirtualiZarr/issues/35).